### PR TITLE
Add an inline-description portion to regex-completion to make it easier to scan for the right choice

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Replacement for https://github.com/dotnet/roslyn/pull/27694

This PR adds an 'inlined description' explaining what the regex item is.  That way you don't have to go examine the full description one at a time.  Very handy for just scanning for the thing you want.  Looks like:

image

image

image

I would still prefer this be right-aligned.  However, maybe we don't need to gate on that?

What do you guys think?

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @jasonmalinowski

**Comment by CyrusNajmabadi** [comment]
Here's a litlte mockup of what right-aligning does.  I def think it's an improvement!

image

File: src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexEmbeddedCompletionProvider.cs [hunk 1]
```diff
@@ -122,6 +122,7 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
                 var sortText = context.Items.Count.ToString("0000");
                 context.AddItem(CompletionItem.Create(
                     displayText: embeddedItem.DisplayText,
+                    inlineDescription: embeddedItem.InlineDescription,
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexEmbeddedCompletionProvider.cs]: We don't have tests on this anywhere?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexEmbeddedCompletionProvider.cs]: nope!

**Comment by CyrusNajmabadi** [comment]
Test added @jasonmalinowski

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski Should we just take this?  It's not like this is a significant deviation from the regex work that went it yesterday.  The only reason this part wasn't included originally was that it was a little hard to read due to the uniform color of the text.  Now that the inline-description is grayed-out, it feels fine to me to go in like this.

You guys can discuss offline if right-aligning would be better than left-aligning.

--

Speaking of which @AmadeusW do you know why the inline description above is not being aligned on the right?  I thought that was something you were doing.  Thanks!

**Comment by jasonmalinowski** [comment]
I'm fine with this as-is, and actually prefer the alignment. Up to you, or @AmadeusW or whoever else has opinions about alignment.

**Comment by CyrusNajmabadi** [comment]
> I'm fine with this as-is, and actually prefer the alignment. Up to you

I'm also ok with it.  Feel free to merge :)

**Comment by jasonmalinowski** [comment]
Sounds good. @dotnet/roslyn-ide any other reviews or opinions? I've kicked integration tests since those are actually working quite reliably now.

**Pull request merged**